### PR TITLE
High Level Metadata Docs

### PR DESCRIPTION
**Pull request by HCastano**
This PR adds some high level descriptions of the `cargo-contract` and ink! metadata. What
it does _not_ include (but should be done in the future):
- Types expected by the described keys
- How to generate values for keys (e.g, selectors)
- Description of `scale-info` types

### CODE REVIEW

File: docs/basics/metadata.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        - `label`: The label of the event.
+        - `args`: The event arguments.
+        - `docs`: The event documentation.
+    - `docs`: The contract documentation.
+
+Note that while all these keys are required, they may be empty. For example, if a
+contract does not define any events then the `events` key would contain an empty array
+`[]`.
+
+#### `storage`
```
Comment (ascjones) [inline docs/basics/metadata.md:142]: Is this documenting the existing or the new storage layout metadata which will be populated my Green's storage rework PR? It should document the new format (or leave it out until that PR is merged), not sure how different it is

**Comment by cmichi** [comment]
@HCastano What is missing for putting this PR into "Ready for Review"?

**Comment by HCastano** [comment]
The broken link is caused by a reference to the `metadata` page, which won't exist until the PR gets merged

**Review by cmichi — APPROVED** [review]
Thank you!

**Review by ascjones — APPROVED** [review]
LGTM! Thanks for doing this :pray:

**Pull request merged**